File: 20-ValidParentheses/20-ValidParentheses.java
// Last updated: 9/8/2025, 4:38:39 PM
class Solution {
    public boolean isValid(String s) {
        for (int i = 1; i < s.length(); i++){
            if (s.substring(i-1, i+1).compareTo("()") == 0 || s.substring(i-1, i+1).compareTo("{}") == 0 || s.substring(i-1, i+1).compareTo("[]") == 0){
                return isValid(s.substring(0, i-1) + s.substring(i+1, s.length()));
            }
        }
        if (s.length() != 0){
            return false;
        }
        else {
            return true;
        }
    }
}

File: 20-ValidParentheses/20-ValidParentheses.py
# Last updated: 2/16/2026, 9:09:31 PM
1class Solution:
2    def isValid(self, s: str) -> bool:
3        pairs = {"()", "[]", "{}"}
4        res = s
5        i = 0
6        while res != "":
7            if i > len(res) - 1:
8                break
9            if res[i:i+2] in pairs:
10                if i is 0:
11                    res = res[i+2:len(res)]
12                elif i is len(res)-2:
13                    res = res[0:i]
14                else:
15                    res = res[0:i] + res[i+2:len(res)]
16                i = 0
17            else:
18                i += 1
19        if res == "":
20            return True
21        return False